### ISSUE TITLE
fix(quant): NaN logits evade the QuaRot forward-equivalence tolerance gate

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#1081]
## Evidence

The QuaRot forward-equivalence chain probe tracks its max error with an IEEE `>` comparison that a NaN silently escapes (`crates/inference/src/quant/quarot/forward_equivalence.rs:317-341`):

```rust
for (a, b) in logits_orig.iter().zip(logits_rot.iter()) {
    let d = (a - b).abs();
    if d > chain_max_abs {   // NaN > x is always false
        chain_max_abs = d;
    }
    chain_total_abs += d;
    chain_count += 1;
}
...
if chain_max_abs > forward_cfg.tolerance {   // stays low if a NaN was skipped
    return Err(...);   // refusal never fires
}
```

If any `a` or `b` is NaN, `d` is NaN, `NaN > chain_max_abs` is false, so `chain_max_abs` is not updated. If it stays below tolerance, the refusal gate passes despite NaN in the produced logits.

## Impact

This gate's entire purpose is to refuse to write conversion artifacts when the rotated pipeline disagrees with the original model. A NaN — the worst possible disagreement — evades it, so corrupt Q4/KHF1 artifacts can be written and shipped. This is a fail-open on the exact failure mode the gate exists to catch.

## Fix shape

Fail closed on any non-finite difference before the tolerance comparison:

```rust
if !d.is_finite() {
    return Err(InferenceError::Inference(format!(
        "forward-equivalence refused: non-finite probe error (a={a}, b={b})"
    )));
}
```

Note `f64::max` would not fix this — `f64::max(NaN, x) == x` also drops the NaN. An explicit `is_finite` check is required. Apply the same guard to the per-tensor check below (`:559-566`).

## Regression test

Inject a NaN into one probe logit: the equivalence check must return `Err` (currently passes). Must fail if the `is_finite` guard is removed.
